src/api/firebase.js

```diff
@@ -235,3 +235,56 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+export function comparePurchaseUrgency(list) {
+	const currentDate = new Date();
+	const soon = [];
+	const kindOfSoon = [];
+	const notSoon = [];
+	const inactive = [];
+	const overdue = [];
+
+	const sortedList = list.sort((a, b) => {
+		const dateNextPurchasedAsDateA = a.dateNextPurchased?.toDate();
+		const dateNextPurchasedAsDateB = b.dateNextPurchased?.toDate();
+
+		const daysUntilNextPurchaseA = getDaysBetweenDates(
+			currentDate,
+			dateNextPurchasedAsDateA,
+		);
+		const daysUntilNextPurchaseB = getDaysBetweenDates(
+			currentDate,
+			dateNextPurchasedAsDateB,
+		);
+
+		return daysUntilNextPurchaseB > daysUntilNextPurchaseA ? -1 : 1;
+	});
+
+	sortedList.forEach((item) => {
```

Comment:
I like the approach you guys took here. It's interesting to me how you created empty array's for each Urgency and then in the if else statement are just pushing values dependent on what is selected by the user.